src/main/java/Wolox/training/controllers/BookController.java

```diff
@@ -1,16 +1,55 @@
 package Wolox.training.controllers;
 
+import Wolox.training.exceptions.BookNotFoundException;
+import Wolox.training.models.Book;
+import Wolox.training.repositories.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.*;
 
 @Controller
+@RequestMapping("/api/books")
 public class BookController {
     @GetMapping("/greeting")
     public String greeting(@RequestParam(name="name", required=false, defaultValue="World") String name, Model model) {
         model.addAttribute("name", name);
         return "greeting";
     }
 
+    @Autowired
+    private BookRepository bookRepository;
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Book create(@RequestBody Book book) {
+        return bookRepository.save(book);
+    }
+
+    @GetMapping("/{id}")
+    public Book findOne(@PathVariable Long id) {
+        return bookRepository.findById(id)
+                .orElseThrow(BookNotFoundException::new);
+    }
+
+    @PutMapping("/{id}")
+    public Book updateBook(@RequestBody Book book, @PathVariable Long id) {
+        if (book.getId() != id) {
+            throw new BookNotFoundException();
```

Comment:
This should be another exception, as it is a bad request because the ids are not the same.
```suggestion
            throw new BookIdMismatchException();
```